### ISSUE TITLE
LoggingTest failure while building on MacOS ARM 

### ISSUE THREAD

**Issue by fyse-nassar** [aws/aws-sdk-cpp#2642]
### Describe the bug

When I build the sdk on macos arm machine, I consistently get failures for the LoggingTest

```
[ 99%] Built target aws-cpp-sdk-glue
[ 99%] Building CXX object generated/tests/glue-gen-tests/CMakeFiles/glue-gen-tests.dir/RunTests.cpp.o
[100%] Building CXX object generated/tests/glue-gen-tests/CMakeFiles/glue-gen-tests.dir/GlueEndpointProviderTests.cpp.o
[100%] Linking CXX executable aws-cpp-sdk-s3-integration-tests
[100%] Built target aws-cpp-sdk-s3-integration-tests
[100%] Linking CXX executable glue-gen-tests
1.11.142
[==========] 51 tests from 1 test suite ran. (3 ms total)
[  PASSED  ] 51 tests.
[100%] Built target glue-gen-tests
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp:274: Failure
Expected equality of these values:
  expectedStatementCount
    Which is: 2
  loggedStatements.size()
    Which is: 3
Logged statements were:
[INFO] 2023-08-16 08:52:01.555 event-loop [0x16f8b3000] id=0x60000121fed0: main loop started
[FATAL] 2023-08-16 08:52:01.555 aws-c-common [0x1f0f66500] test fatal level
[FATAL] 2023-08-16 08:52:01.555 aws-c-common [0x1f0f66500] test fatal format level
Google Test trace:
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp:314: DoCRTLogTest logLevel: FATAL
[  FAILED  ] CRTLoggingTest.testFatalLogLevel (0 ms)
[==========] 507 tests from 69 test suites ran. (10488 ms total)
[  PASSED  ] 505 tests.
[  SKIPPED ] 1 test.
1.11.142
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp:274: Failure
Expected equality of these values:
  expectedStatementCount
    Which is: 6
  loggedStatements.size()
    Which is: 7
Logged statements were:
[INFO] 2023-08-16 08:52:12.294 event-loop [0x16f93f000] id=0x600001224140: main loop started
[FATAL] 2023-08-16 08:52:12.294 aws-c-common [0x1f0f66500] test fatal level
[FATAL] 2023-08-16 08:52:12.294 aws-c-common [0x1f0f66500] test fatal format level
[ERROR] 2023-08-16 08:52:12.294 aws-c-common [0x1f0f66500] test error level
[ERROR] 2023-08-16 08:52:12.294 aws-c-common [0x1f0f66500] test error format level
[WARN] 2023-08-16 08:52:12.294 aws-c-common [0x1f0f66500] test warn level
[WARN] 2023-08-16 08:52:12.294 aws-c-common [0x1f0f66500] test warn format level
Google Test trace:
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp:314: DoCRTLogTest logLevel: WARN
[  FAILED  ] CRTLoggingTest.testWarnLogLevel (0 ms)
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/memory/AWSMemoryTest.cpp:105: Skipped
Skipping the repeat of StaticUniquePtrDtorOrder test as it cannot be repeated.
[==========] 507 tests from 69 test suites ran. (15334 ms total)
[  PASSED  ] 505 tests.
[  SKIPPED ] 1 test.
make[2]: *** [tests/aws-cpp-sdk-core-tests/aws-cpp-sdk-core-tests] Error 1
make[2]: *** Deleting file `tests/aws-cpp-sdk-core-tests/aws-cpp-sdk-core-tests'
make[1]: *** [tests/aws-cpp-sdk-core-tests/CMakeFiles/aws-cpp-sdk-core-tests.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX executable s3-gen-tests
[==========] 272 tests from 1 test suite ran. (237 ms total)
[  PASSED  ] 272 tests.
[100%] Built target s3-gen-tests
make: *** [all] Error 2
Cleaning up project directory and file based variables
00:00
ERROR: Job failed: exit status 1
``` 

### Expected Behavior

No test failures.

### Current Behavior

Test failing during default build.

### Reproduction Steps

Build the SDK on a MacOS arm machine.

### AWS CPP SDK version used

1.11.142

### Compiler and Version used

AppleClang 14.0.0.14000029

### Operating System and version

MacOS arm64

**Comment by SergeyRyabinin** [comment]
Hi, thank you for opening the issue.
I've fixed one issue within this test literally few versions ago in 1.11.149 in this PR: https://github.com/aws/aws-sdk-cpp/pull/2637

Could you please confirm that you can still observe this issue in a most recent version (min 1.11.149)?

Best regards,
Sergey

**Comment by fyse-nassar** [comment]
Yup, I did see that commit and tested that particular version (1.11.149) too. I still hit the same issue.
```
[ 96%] Building CXX object tests/aws-cpp-sdk-core-tests/CMakeFiles/aws-cpp-sdk-core-tests.dir/utils/xml/XmlSerializerTest.cpp.o
[ 96%] Linking CXX executable aws-cpp-sdk-core-tests
1.11.149
[ 96%] Linking CXX shared library libaws-cpp-sdk-s3.dylib
[ 96%] Built target aws-cpp-sdk-s3
[ 96%] Building CXX object tests/aws-cpp-sdk-s3-integration-tests/CMakeFiles/aws-cpp-sdk-s3-integration-tests.dir/BucketAndObjectOperationTest.cpp.o
[ 96%] Building CXX object generated/tests/s3-gen-tests/CMakeFiles/s3-gen-tests.dir/RunTests.cpp.o
[ 97%] Building CXX object generated/tests/s3-gen-tests/CMakeFiles/s3-gen-tests.dir/S3EndpointProviderTests.cpp.o
[ 98%] Building CXX object tests/aws-cpp-sdk-s3-integration-tests/CMakeFiles/aws-cpp-sdk-s3-integration-tests.dir/RunTests.cpp.o
[ 99%] Linking CXX shared library libaws-cpp-sdk-glue.dylib
[ 99%] Built target aws-cpp-sdk-glue
[ 99%] Building CXX object generated/tests/glue-gen-tests/CMakeFiles/glue-gen-tests.dir/RunTests.cpp.o
[100%] Building CXX object generated/tests/glue-gen-tests/CMakeFiles/glue-gen-tests.dir/GlueEndpointProviderTests.cpp.o
[100%] Linking CXX executable aws-cpp-sdk-s3-integration-tests
[100%] Built target aws-cpp-sdk-s3-integration-tests
[100%] Linking CXX executable glue-gen-tests
[==========] 51 tests from 1 test suite ran. (3 ms total)
[  PASSED  ] 51 tests.
[100%] Built target glue-gen-tests
[==========] 512 tests from 70 test suites ran. (28429 ms total)
[  PASSED  ] 512 tests.
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/memory/AWSMemoryTest.cpp:105: Skipped
Skipping the repeat of StaticUniquePtrDtorOrder test as it cannot be repeated.
1.11.149
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/logging/LoggingTest.cpp:196: Failure
Expected equality of these values:
  expectedStatementCount
    Which is: 3
  loggedStatements.size()
    Which is: 6
Logged statements were:
[FATAL] 2023-08-24 07:24:02.849 LoggingTest_testFatalLogLevel [0x1f0f66500] test fatal level
[INFO] 2023-08-24 07:24:02.849 event-loop [0x16debb000] id=0x600002958000: main loop started
[FATAL] 2023-08-24 07:24:02.849 LoggingTest_testFatalLogLevel [0x1f0f66500] test fatal format level
[INFO] 2023-08-24 07:24:02.849 event-loop [0x16debb000] id=0x600002958000: default timeout 100s, and max events to process per tick 100
[FATAL] 2023-08-24 07:24:02.849 LoggingTest_testFatalLogLevel [0x1f0f66500] test fatal stream level
[TRACE] 2023-08-24 07:24:02.849 event-loop [0x16debb000] id=0x600002958000: waiting for a maximum of 100s 0ns
[  FAILED  ] LoggingTest.testFatalLogLevel (350 ms)
[==========] 512 tests from 70 test suites ran. (25199 ms total)
[  PASSED  ] 510 tests.
[  SKIPPED ] 1 test.
1.11.149
/Users/dp/builds/fnc8zzy5/0/dp/mirrors/build-aws-sdk-cpp/aws-sdk-cpp/tests/aws-cpp-sdk-core-tests/utils/memory/AWSMemoryTest.cpp:105: Skipped
Skipping the repeat of StaticUniquePtrDtorOrder test as it cannot be repeated.
[==========] 512 tests from 70 test suites ran. (21047 ms total)
[  PASSED  ] 511 tests.
[  SKIPPED ] 1 test.
make[2]: *** [tests/aws-cpp-sdk-core-tests/aws-cpp-sdk-core-tests] Error 1
make[2]: *** Deleting file `tests/aws-cpp-sdk-core-tests/aws-cpp-sdk-core-tests'
make[1]: *** [tests/aws-cpp-sdk-core-tests/CMakeFiles/aws-cpp-sdk-core-tests.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX executable s3-gen-tests
[==========] 272 tests from 1 test suite ran. (236 ms total)
[  PASSED  ] 272 tests.
[100%] Built target s3-gen-tests
make: *** [all] Error 2
Cleaning up project directory and file based variables
00:00
ERROR: Job failed: exit status 1
```

**Comment by SergeyRyabinin** [comment]
Hi @fyse-nassar,

> Was the fix specific to Windows? I observe this on MacOS arm machine.

It appeared to be a different issue.

Thank you for reporting this issue and confirming it still occurred. 
I'm not sure why our testing did not catch this, I've updated the test to not fail on this case (when CRT event loop logs something on restart and it happens during CRT logging test execution).
It shall be available in #[1.11.153](https://github.com/aws/aws-sdk-cpp/releases/tag/1.11.153) after today's daily release.

Please reopen this issue or submit a new one shall you see one or very similar issue again.

Best regards,
Sergey
